app/models/application_record.rb

```diff
@@ -1,3 +1,5 @@
 class ApplicationRecord < ActiveRecord::Base
+  strip_attributes allow_empty: true
+
```

Comment:
allow_empty is true because some forms and models depend on empty string defaults. I thought about putting it on specific models, but without fantaaastic spec coverage or a much more meticulous pass at this, I think there's too high a risk of regression bugs that Madeline staff would see as internal errors, in areas of the app that are well-trodden.

Reply:
upon closer look (thanks for the prompt @meltheadorable ) all these errors were coming from translatable.html.slim:
```1) basic project flow can index/show/edit and change division
     Failure/Error: = project.summary
     
     ActionView::Template::Error:
       undefined method `scan' for nil:NilClass
     Shared Example Group: :flow called from ./spec/features/admin/basic_project_flow_spec.rb:15
     # ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/projects/_translatable.html.slim:2:in `_app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/basic_projects/_form.html.slim:25:in `block in _app_views_admin_basic_projects__form_html_slim__3083474540500009643_70155474720560'
     # ./app/views/admin/basic_projects/_form.html.slim:1:in `_app_views_admin_basic_projects__form_html_slim__3083474540500009643_70155474720560'
     # ./app/views/admin/basic_projects/_details.slim:19:in `_app_views_admin_basic_projects__details_slim__4340730682223372658_70155485889400'
     # ./app/views/admin/basic_projects/show.html.slim:10:in `_app_views_admin_basic_projects_show_html_slim__4165249263214135805_70155486183560'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `scan' for nil:NilClass
     #   ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'

  2) basic project flow validations for creating basic project
     Failure/Error: = project.summary
     
     ActionView::Template::Error:
       undefined method `scan' for nil:NilClass
     # ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/projects/_translatable.html.slim:2:in `_app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/basic_projects/_form.html.slim:25:in `block in _app_views_admin_basic_projects__form_html_slim__3083474540500009643_70155474720560'
     # ./app/views/admin/basic_projects/_form.html.slim:1:in `_app_views_admin_basic_projects__form_html_slim__3083474540500009643_70155474720560'
     # ./app/views/admin/basic_projects/new.html.slim:5:in `_app_views_admin_basic_projects_new_html_slim___3401588978841990_70155482005600'
     # ./app/controllers/admin/basic_projects_controller.rb:76:in `create'
     # ./spec/features/admin/basic_project_flow_spec.rb:24:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `scan' for nil:NilClass
     #   ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'

  3) basic project flow validations for updating basic project
     Failure/Error: = project.summary
     
     ActionView::Template::Error:
       undefined method `scan' for nil:NilClass
     # ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/projects/_translatable.html.slim:2:in `_app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/basic_projects/_form.html.slim:25:in `block in _app_views_admin_basic_projects__form_html_slim__3083474540500009643_70155474720560'
     # ./app/views/admin/basic_projects/_form.html.slim:1:in `_app_views_admin_basic_projects__form_html_slim__3083474540500009643_70155474720560'
     # ./app/views/admin/basic_projects/_details.slim:19:in `_app_views_admin_basic_projects__details_slim__4340730682223372658_70155485889400'
     # ./app/views/admin/basic_projects/show.html.slim:10:in `_app_views_admin_basic_projects_show_html_slim__4165249263214135805_70155486183560'
     # ./app/controllers/admin/basic_projects_controller.rb:61:in `update'
     # ./spec/features/admin/basic_project_flow_spec.rb:33:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `scan' for nil:NilClass
     #   ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'

  4) date picker i18n english should have visible css ".loan_signing_date" with text "Aug 19, 2017"
     Failure/Error: = project.summary
     
     ActionView::Template::Error:
       undefined method `scan' for nil:NilClass
     # ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/projects/_translatable.html.slim:2:in `_app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/loans/_form.html.slim:77:in `block in _app_views_admin_loans__form_html_slim___973625804511057208_70155441543060'
     # ./app/views/admin/loans/_form.html.slim:1:in `_app_views_admin_loans__form_html_slim___973625804511057208_70155441543060'
     # ./app/views/admin/loans/_details.html.slim:54:in `_app_views_admin_loans__details_html_slim__3739508125684967512_70155394572400'
     # ./app/views/admin/loans/show.html.slim:15:in `_app_views_admin_loans_show_html_slim__2634205714876616608_70155492081280'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `scan' for nil:NilClass
     #   ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'

  5) date picker i18n spanish should have visible css ".loan_signing_date" with text "ago 19, 2017"
     Failure/Error: = project.summary
     
     ActionView::Template::Error:
       undefined method `scan' for nil:NilClass
     # ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155480809680'
     # ./app/views/admin/projects/_translatable.html.slim:2:in `_app_views_admin_projects__translatable_html_slim__1567477515628480703_70155480809680'
     # ./app/views/admin/loans/_form.html.slim:77:in `block in _app_views_admin_loans__form_html_slim___973625804511057208_70155482106580'
     # ./app/views/admin/loans/_form.html.slim:1:in `_app_views_admin_loans__form_html_slim___973625804511057208_70155482106580'
     # ./app/views/admin/loans/_details.html.slim:54:in `_app_views_admin_loans__details_html_slim__3739508125684967512_70155482694780'
     # ./app/views/admin/loans/show.html.slim:15:in `_app_views_admin_loans_show_html_slim__2634205714876616608_70155473553820'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `scan' for nil:NilClass
     #   ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155480809680'

  6) loan flow can index/show/edit and change division
     Failure/Error: = project.summary
     
     ActionView::Template::Error:
       undefined method `scan' for nil:NilClass
     Shared Example Group: :flow called from ./spec/features/admin/loan_flow_spec.rb:15
     # ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/projects/_translatable.html.slim:2:in `_app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/loans/_form.html.slim:77:in `block in _app_views_admin_loans__form_html_slim___973625804511057208_70155441543060'
     # ./app/views/admin/loans/_form.html.slim:1:in `_app_views_admin_loans__form_html_slim___973625804511057208_70155441543060'
     # ./app/views/admin/loans/_details.html.slim:54:in `_app_views_admin_loans__details_html_slim__3739508125684967512_70155394572400'
     # ./app/views/admin/loans/show.html.slim:15:in `_app_views_admin_loans_show_html_slim__2634205714876616608_70155492081280'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `scan' for nil:NilClass
     #   ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'

  7) loan flow loan can not be created with the same person as pry and sec agent
     Failure/Error: = project.summary
     
     ActionView::Template::Error:
       undefined method `scan' for nil:NilClass
     # ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/projects/_translatable.html.slim:2:in `_app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'
     # ./app/views/admin/loans/_form.html.slim:77:in `block in _app_views_admin_loans__form_html_slim___973625804511057208_70155494874880'
     # ./app/views/admin/loans/_form.html.slim:1:in `_app_views_admin_loans__form_html_slim___973625804511057208_70155494874880'
     # ./app/views/admin/loans/new.html.slim:6:in `_app_views_admin_loans_new_html_slim__3288383588683714429_70155474886440'
     # ./app/controllers/admin/loans_controller.rb:102:in `create'
     # ./spec/features/admin/loan_flow_spec.rb:77:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `scan' for nil:NilClass
     #   ./app/views/admin/projects/_translatable.html.slim:3:in `block in _app_views_admin_projects__translatable_html_slim__1567477515628480703_70155489324060'

Finished in 3 minutes 22.9 seconds (files took 10.46 seconds to load)
1013 examples, 7 failures, 49 pending

Failed examples:

rspec ./spec/features/admin/basic_project_flow_spec.rb:3 # basic project flow can index/show/edit and change division
rspec ./spec/features/admin/basic_project_flow_spec.rb:20 # basic project flow validations for creating basic project
rspec ./spec/features/admin/basic_project_flow_spec.rb:28 # basic project flow validations for updating basic project
rspec ./spec/features/admin/date_picker_i18n_spec.rb:16 # date picker i18n english should have visible css ".loan_signing_date" with text "Aug 19, 2017"
rspec ./spec/features/admin/date_picker_i18n_spec.rb:31 # date picker i18n spanish should have visible css ".loan_signing_date" with text "ago 19, 2017"
rspec ./spec/features/admin/loan_flow_spec.rb:3 # loan flow can index/show/edit and change division
rspec ./spec/features/admin/loan_flow_spec.rb:73 # loan flow loan can not be created with the same person as pry and sec agent```